### PR TITLE
Fix multi python package publishing

### LINKED ISSUE #200: Packages uploaded multiple times
I just updated my fork of jupyter_releaser to the latest, and trying to release jupyverse [fails](https://github.com/davidbrochart/jupyter_releaser/runs/4090888225?check_suite_focus=true). It looks like it's uploading the packages again and again. For instance, `fps-auth` was uploaded several times, and I can see that the version is there on PyPI, but then we get a `403`.
I'm not sure why it is uploaded several times, any idea?

### PR DESCRIPTION
**Pull request by davidbrochart**
We now build Python packages in separate distribution directories, and upload them to PyPI independently. This also means publishing assets to PyPI and to NPM separately.

Fixes #200

### CODE REVIEW

**Comment by davidbrochart** [comment]
The problem is that no matter if we build packages in separate directories, we install them in a common directory before uploading them to PyPI, so it's useless.
What I did is to provide a list of packages in the form `"package_path:package_name"`, like so:
https://github.com/jupyter-server/jupyverse/blob/d02cde63d0bf2e47fb5d9310b74a9fc6c95b9ce1/pyproject.toml#L22
I added the `package_name` because we need the package name to map the distribution files to their PyPI token.

File: jupyter_releaser/lib.py [hunk 1]
```diff
@@ -413,12 +420,15 @@ def publish_assets(
...
         if suffix in [".gz", ".whl"]:
-            env = os.environ.copy()
-            env["TWINE_PASSWORD"] = twine_token
-            # NOTE: Do not print the env since a twine token extracted from
-            # a PYPI_TOKEN_MAP will not be sanitized in output
-            util.retry(f"{twine_cmd} {name}", cwd=dist_dir, env=env)
-            found = True
+            if name.startswith(
+                python_package_name
+            ):  # FIXME: not enough to know it's the right package
```
Comment (davidbrochart) [inline jupyter_releaser/lib.py:425]: The distribution file name is something like `my_package-0.1.2.tar.gz` or `my_package-0.1.2-py3-none-any.whl`, and we could imagine a package name being `my_package` and another one being `my_package_2`, and both would match this condition, so this is not great. Maybe we could include the version spec in the condition, but at first sight tests were failing when doing so.

**Comment by jtpio** [comment]
Thanks @davidbrochart for looking into this.

File: .github/actions/check-links/action.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ runs:
   using: "composite"
   steps:
     - name: install-releaser
-      uses: jupyter-server/jupyter_releaser/.github/actions/install-releaser@v1
+      uses: davidbrochart/jupyter_releaser/.github/actions/install-releaser@fix_multipy
```
Comment (jtpio) [inline .github/actions/check-links/action.yml:7]: We'll need to revert that one (and similar below) to point to the releaser actions.

**Comment by davidbrochart** [comment]
This is ready to be merged, and jupyverse depends on it. Given that the multi-python packages feature is only used by jupyverse AFAIK, I think the current limitation is acceptable, and we can improve it in the future. The limitation is that the package names, given in the form:
```
python_packages = [".:jupyverse", "plugins/auth:fps-auth", "plugins/contents:fps-contents"]
```
must not start with the same name, e.g. something like that wouldn't work:
```
python_packages = [".:jupyverse", "plugins/auth:fps-auth", "plugins/contents:fps-auth2"]
```

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**